airbyte-integrations/connectors/destination-mssql/src/main/kotlin/io/airbyte/integrations/destination/mssql/v2/MSSQLQueryBuilder.kt

```diff
@@ -338,59 +323,44 @@ class MSSQLQueryBuilder(
         plainRecord: DestinationRecord,
         schema: List<NamedField>
     ) {
-        val record = plainRecord.asRecordMarshaledToAirbyteValue()
-        val recordObject = record.data as ObjectValue
-        var airbyteMetaStatementIndex: Int? = null
-        val airbyteMeta =
-            AirbyteRecordMessageMeta().apply {
-                changes =
-                    record.meta?.changes?.map { it.asProtocolObject() }?.toMutableList()
-                        ?: mutableListOf()
-                setAdditionalProperty("sync_id", stream.syncId)
-            }
+        val enrichedRecord =
+            plainRecord.asDestinationRecordRaw().asEnrichedDestinationRecordAirbyteValue()
+        // TODO maybe we should just sort this into schema order?
+        val populatedFields = enrichedRecord.allTypedFields
 
+        var airbyteMetaStatementIndex: Int? = null
         schema.forEachIndexed { index, field ->
             val statementIndex = index + 1
-            if (field.name in airbyteFields) {
-                when (field.name) {
-                    COLUMN_NAME_AB_RAW_ID ->
-                        statement.setString(statementIndex, UUID.randomUUID().toString())
-                    COLUMN_NAME_AB_EXTRACTED_AT ->
-                        statement.setLong(statementIndex, record.emittedAtMs)
-                    COLUMN_NAME_AB_GENERATION_ID ->
-                        statement.setLong(statementIndex, stream.generationId)
-                    COLUMN_NAME_AB_META -> airbyteMetaStatementIndex = statementIndex
-                }
-            } else {
+            if (field.name == COLUMN_NAME_AB_META) {
+                // don't populate _airbyte_meta yet - we might run into errors in the other fields
```

Comment:
this isn't new, it matches the previous behavior (`COLUMN_NAME_AB_META -> airbyteMetaStatementIndex = statementIndex`). Just adding a comment b/c it's nonobvious.